integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/RegexpPredicateBaseIT.java

```diff
@@ -0,0 +1,445 @@
+				SimpleMappedIndex.of( root -> new IndexBinding( root, supportedFieldTypes ) )
+						.name( "argumentChecking" );
+
+		@Parameterized.Parameters(name = "{0}")
+		public static List<Object[]> parameters() {
+			return parameters;
+		}
+
+		public ArgumentCheckingIT(FieldTypeDescriptor<?> fieldType) {
+			super( index, fieldType );
+		}
+
+		@Override
+		protected void tryPredicateWithNullMatchingParam(SearchPredicateFactory f, String fieldPath) {
+			f.regexp().field( fieldPath ).matching( null );
+		}
+	}
+
+	@RunWith(Parameterized.class)
+	public static class TypeCheckingNoConversionIT
+			extends AbstractPredicateTypeCheckingNoConversionIT<RegexpPredicateTestValues> {
+		private static final List<DataSet<String, RegexpPredicateTestValues>> dataSets = new ArrayList<>();
+		private static final List<Object[]> parameters = new ArrayList<>();
+		static {
+			for ( FieldTypeDescriptor<String> fieldType : supportedFieldTypes ) {
+				DataSet<String, RegexpPredicateTestValues> dataSet = new DataSet<>( testValues( fieldType ) );
+				dataSets.add( dataSet );
+				parameters.add( new Object[] { dataSet } );
+			}
+		}
+
+		@Parameterized.Parameters(name = "{0}")
+		public static List<Object[]> parameters() {
+			return parameters;
+		}
+
+		private static final SimpleMappedIndex<IndexBinding> index =
+				SimpleMappedIndex.of( root -> new IndexBinding( root, supportedFieldTypes ) )
+						.name( "typeChecking_main" );
+		private static final SimpleMappedIndex<CompatibleIndexBinding> compatibleIndex =
+				SimpleMappedIndex.of( root -> new CompatibleIndexBinding( root, supportedFieldTypes ) )
+						.name( "typeChecking_compatible" );
+		private static final SimpleMappedIndex<RawFieldCompatibleIndexBinding> rawFieldCompatibleIndex =
+				SimpleMappedIndex.of( root -> new RawFieldCompatibleIndexBinding( root, supportedFieldTypes ) )
+						.name( "typeChecking_rawFieldCompatible" );
+		private static final SimpleMappedIndex<IncompatibleIndexBinding> incompatibleIndex =
+				SimpleMappedIndex.of( root -> new IncompatibleIndexBinding( root, supportedFieldTypes ) )
+						.name( "typeChecking_incompatible" );
+
+		public TypeCheckingNoConversionIT(DataSet<String, RegexpPredicateTestValues> dataSet) {
+			super( index, compatibleIndex, rawFieldCompatibleIndex, incompatibleIndex, dataSet );
+		}
+
+		@Override
+		protected PredicateFinalStep predicate(SearchPredicateFactory f, String fieldPath, int matchingDocOrdinal) {
+			return f.regexp().field( fieldPath ).matching( dataSet.values.matchingArg( matchingDocOrdinal ) );
+		}
+
+		@Override
+		protected PredicateFinalStep predicate(SearchPredicateFactory f, String field0Path, String field1Path,
+				int matchingDocOrdinal) {
+			return f.regexp().field( field0Path ).field( field1Path )
+					.matching( dataSet.values.matchingArg( matchingDocOrdinal ) );
+		}
+
+		@Override
+		protected String predicateNameInErrorMessage() {
+			return "predicate:regexp";
+		}
+	}
+}
```

Comment:
These tests are only using very exotic regular expressions, bascially just literals. That's fine, because these tests are not about regexps but all the other features. But what about testing actual regexps?

Please add some tests with a few *real* regular expressions. You can copy `org.hibernate.search.integrationtest.backend.tck.search.predicate.WildcardPredicateSpecificsIT`, which by the way may have some tests that are relevant for the regexp predicate too, and add a few test methods.